engineioxide/src/protocol.rs

```diff
@@ -0,0 +1,21 @@
+use std::str::FromStr;
+
+use crate::errors::Error;
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum ProtocolVersion {
+    V3 = 3,
+    V4 = 4,
+}
+
+impl FromStr for ProtocolVersion {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "3" => Ok(ProtocolVersion::V3),
+            "4" => Ok(ProtocolVersion::V4),
+            _ => Err(Error::UnsupportedProtocolVersion),
+        }
+    }
+}
```

Comment:
It is better to put the `ProtocolVersion` struct next to the `TransportType` in the `service` mod since they are both of the same "kind"

Reply:
`Copy` can be derived for this enum, it will avoid explicit `clone` calls